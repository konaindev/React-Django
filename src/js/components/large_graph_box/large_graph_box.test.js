import renderer from "react-test-renderer";
import { CurrencyShorthandGraphBox, PercentageGraphBox } from "./index";

describe("LargeGraphBox", () => {
  beforeEach(() => {
    Math.random = jest.fn(() => "12345");
  });

  it("renders PercentageGraphBox correctly", () => {
    const props = {
      name: "USV > EXE",
      value: 0.1,
      target: 0.13,
      delta: 0.03,
      series: [10, 20, 30, 15]
    };
    const tree = renderer.create(<PercentageGraphBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders PercentageGraphBox correctly with null target value", () => {
    const props = {
      name: "USV > EXE",
      value: 0.1,
      target: null,
      delta: 0.03,
      series: [10, 20, 30, 15]
    };
    const tree = renderer.create(<PercentageGraphBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders PercentageGraphBox with extra content correctly", () => {
    const props = {
      name: "USV > EXE",
      value: 0.1,
      target: 0.13,
      delta: 0.03,
      extraContent: "227 Executed Leases (Out of 260)",
      series: [10, 20, 30, 15]
    };
    const tree = renderer.create(<PercentageGraphBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders CurrencyShorthandGraphBox correctly", () => {
    const props = {
      name: "USV > EXE",
      value: 13456,
      target: 32423,
      delta: 1232,
      series: [10, 20, 30, 15]
    };

    const tree = renderer.create(<CurrencyShorthandGraphBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders CurrencyShorthandGraphBox with extra content correctly", () => {
    const props = {
      name: "USV > EXE",
      value: 12345,
      target: 32343,
      delta: 132,
      extraContent: "227 Executed Leases (Out of 260)",
      series: [10, 20, 30, 15]
    };
    const tree = renderer.create(<CurrencyShorthandGraphBox {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
