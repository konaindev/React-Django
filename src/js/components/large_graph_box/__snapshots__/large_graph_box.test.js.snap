// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LargeGraphBox renders CurrencyShorthandGraphBox correctly 1`] = `
<div
  className="panel panel--default large-box"
>
  <span
    className="large-box__top-line"
  >
    USV &gt; EXE
  </span>
  <div
    className="large-box__content"
  >
    <span
      className="large-box__content-value"
    >
      $13.5k
    </span>
    <div
      className="large-box__content-extra"
    >
      <div
        className="large-box__content-graph"
      >
        <div
          className="VictoryContainer"
          style={
            Object {
              "height": "100%",
              "pointerEvents": "none",
              "position": "relative",
              "touchAction": "none",
              "width": "100%",
            }
          }
        >
          <svg
            aria-labelledby="victory-container-5-title victory-container-5-desc"
            height={300}
            role="img"
            style={
              Object {
                "height": "100%",
                "pointerEvents": "all",
                "width": "100%",
              }
            }
            viewBox="0 0 450 300"
            width={450}
          >
            <defs>
              <linearGradient
                id="-49270000"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stopColor="#65FF00"
                  stopOpacity={1}
                />
                <stop
                  offset="19%"
                  stopColor="#65FF00"
                  stopOpacity={0.738}
                />
                <stop
                  offset="34%"
                  stopColor="#65FF00"
                  stopOpacity={0.541}
                />
                <stop
                  offset="47%"
                  stopColor="#65FF00"
                  stopOpacity={0.382}
                />
                <stop
                  offset="57%"
                  stopColor="#65FF00"
                  stopOpacity={0.278}
                />
                <stop
                  offset="65%"
                  stopColor="#65FF00"
                  stopOpacity={0.194}
                />
                <stop
                  offset="73%"
                  stopColor="#65FF00"
                  stopOpacity={0.126}
                />
                <stop
                  offset="80%"
                  stopColor="#65FF00"
                  stopOpacity={0.075}
                />
                <stop
                  offset="86%"
                  stopColor="#65FF00"
                  stopOpacity={0.042}
                />
                <stop
                  offset="91%"
                  stopColor="#65FF00"
                  stopOpacity={0.021}
                />
                <stop
                  offset="95%"
                  stopColor="#65FF00"
                  stopOpacity={0.008}
                />
                <stop
                  offset="98%"
                  stopColor="#65FF00"
                  stopOpacity={0.002}
                />
                <stop
                  offset="100%"
                  stopColor="#65FF00"
                  stopOpacity={0}
                />
              </linearGradient>
            </defs>
            <g
              clipPath="url(#victory-clip-6)"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <defs>
                <clipPath
                  id="victory-clip-6"
                >
                  <rect
                    height={300}
                    role="presentation"
                    shapeRendering="auto"
                    vectorEffect="non-scaling-stroke"
                    width={450}
                    x={0}
                    y={0}
                  />
                </clipPath>
              </defs>
              <g>
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150L450,300L425,300C400,300,350,300,300,300C250,300,200,300,150,300C100,300,50,300,25,300L0,300Z"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "url(#-49270000)",
                      "stroke": "none",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "none",
                      "stroke": "#65FF00",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
              </g>
            </g>
          </svg>
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
                "zIndex": 99,
              }
            }
          >
            <svg
              height={300}
              style={
                Object {
                  "height": "100%",
                  "overflow": "visible",
                  "width": "100%",
                }
              }
              viewBox="0 0 450 300"
              width={450}
            />
          </div>
        </div>
      </div>
      <div
        className="large-box__content-delta"
      >
        <div
          className="delta-indicator"
        >
          <span
            className="delta-indicator__delta"
          >
            $1.2k
          </span>
           
          <span
            className="delta-indicator__arrow delta-indicator__trend-up"
          >
            <svg
              aria-hidden="true"
              color="inherit"
              focusable="false"
              height={5}
              role="presentation"
              viewBox="0 0 10 5"
              width={10}
            >
              <path
                d="M5 0l5 5H0z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <p
    className="large-box__bottom-line"
  />
  <p
    className="large-box__bottom-line"
  >
    Target: $32.4k
  </p>
</div>
`;

exports[`LargeGraphBox renders CurrencyShorthandGraphBox with extra content correctly 1`] = `
<div
  className="panel panel--default large-box"
>
  <span
    className="large-box__top-line"
  >
    USV &gt; EXE
  </span>
  <div
    className="large-box__content"
  >
    <span
      className="large-box__content-value"
    >
      $12.3k
    </span>
    <div
      className="large-box__content-extra"
    >
      <div
        className="large-box__content-graph"
      >
        <div
          className="VictoryContainer"
          style={
            Object {
              "height": "100%",
              "pointerEvents": "none",
              "position": "relative",
              "touchAction": "none",
              "width": "100%",
            }
          }
        >
          <svg
            aria-labelledby="victory-container-7-title victory-container-7-desc"
            height={300}
            role="img"
            style={
              Object {
                "height": "100%",
                "pointerEvents": "all",
                "width": "100%",
              }
            }
            viewBox="0 0 450 300"
            width={450}
          >
            <defs>
              <linearGradient
                id="-49270000"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stopColor="#65FF00"
                  stopOpacity={1}
                />
                <stop
                  offset="19%"
                  stopColor="#65FF00"
                  stopOpacity={0.738}
                />
                <stop
                  offset="34%"
                  stopColor="#65FF00"
                  stopOpacity={0.541}
                />
                <stop
                  offset="47%"
                  stopColor="#65FF00"
                  stopOpacity={0.382}
                />
                <stop
                  offset="57%"
                  stopColor="#65FF00"
                  stopOpacity={0.278}
                />
                <stop
                  offset="65%"
                  stopColor="#65FF00"
                  stopOpacity={0.194}
                />
                <stop
                  offset="73%"
                  stopColor="#65FF00"
                  stopOpacity={0.126}
                />
                <stop
                  offset="80%"
                  stopColor="#65FF00"
                  stopOpacity={0.075}
                />
                <stop
                  offset="86%"
                  stopColor="#65FF00"
                  stopOpacity={0.042}
                />
                <stop
                  offset="91%"
                  stopColor="#65FF00"
                  stopOpacity={0.021}
                />
                <stop
                  offset="95%"
                  stopColor="#65FF00"
                  stopOpacity={0.008}
                />
                <stop
                  offset="98%"
                  stopColor="#65FF00"
                  stopOpacity={0.002}
                />
                <stop
                  offset="100%"
                  stopColor="#65FF00"
                  stopOpacity={0}
                />
              </linearGradient>
            </defs>
            <g
              clipPath="url(#victory-clip-8)"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <defs>
                <clipPath
                  id="victory-clip-8"
                >
                  <rect
                    height={300}
                    role="presentation"
                    shapeRendering="auto"
                    vectorEffect="non-scaling-stroke"
                    width={450}
                    x={0}
                    y={0}
                  />
                </clipPath>
              </defs>
              <g>
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150L450,300L425,300C400,300,350,300,300,300C250,300,200,300,150,300C100,300,50,300,25,300L0,300Z"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "url(#-49270000)",
                      "stroke": "none",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "none",
                      "stroke": "#65FF00",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
              </g>
            </g>
          </svg>
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
                "zIndex": 99,
              }
            }
          >
            <svg
              height={300}
              style={
                Object {
                  "height": "100%",
                  "overflow": "visible",
                  "width": "100%",
                }
              }
              viewBox="0 0 450 300"
              width={450}
            />
          </div>
        </div>
      </div>
      <div
        className="large-box__content-delta"
      >
        <div
          className="delta-indicator"
        >
          <span
            className="delta-indicator__delta"
          >
            $132
          </span>
           
          <span
            className="delta-indicator__arrow delta-indicator__trend-up"
          >
            <svg
              aria-hidden="true"
              color="inherit"
              focusable="false"
              height={5}
              role="presentation"
              viewBox="0 0 10 5"
              width={10}
            >
              <path
                d="M5 0l5 5H0z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <p
    className="large-box__bottom-line"
  >
    227 Executed Leases (Out of 260)
  </p>
  <p
    className="large-box__bottom-line"
  >
    Target: $32.3k
  </p>
</div>
`;

exports[`LargeGraphBox renders PercentageGraphBox correctly 1`] = `
<div
  className="panel panel--default large-box"
>
  <span
    className="large-box__top-line"
  >
    USV &gt; EXE
  </span>
  <div
    className="large-box__content"
  >
    <span
      className="large-box__content-value"
    >
      10%
    </span>
    <div
      className="large-box__content-extra"
    >
      <div
        className="large-box__content-graph"
      >
        <div
          className="VictoryContainer"
          style={
            Object {
              "height": "100%",
              "pointerEvents": "none",
              "position": "relative",
              "touchAction": "none",
              "width": "100%",
            }
          }
        >
          <svg
            aria-labelledby="victory-container-1-title victory-container-1-desc"
            height={300}
            role="img"
            style={
              Object {
                "height": "100%",
                "pointerEvents": "all",
                "width": "100%",
              }
            }
            viewBox="0 0 450 300"
            width={450}
          >
            <defs>
              <linearGradient
                id="-49270000"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stopColor="#65FF00"
                  stopOpacity={1}
                />
                <stop
                  offset="19%"
                  stopColor="#65FF00"
                  stopOpacity={0.738}
                />
                <stop
                  offset="34%"
                  stopColor="#65FF00"
                  stopOpacity={0.541}
                />
                <stop
                  offset="47%"
                  stopColor="#65FF00"
                  stopOpacity={0.382}
                />
                <stop
                  offset="57%"
                  stopColor="#65FF00"
                  stopOpacity={0.278}
                />
                <stop
                  offset="65%"
                  stopColor="#65FF00"
                  stopOpacity={0.194}
                />
                <stop
                  offset="73%"
                  stopColor="#65FF00"
                  stopOpacity={0.126}
                />
                <stop
                  offset="80%"
                  stopColor="#65FF00"
                  stopOpacity={0.075}
                />
                <stop
                  offset="86%"
                  stopColor="#65FF00"
                  stopOpacity={0.042}
                />
                <stop
                  offset="91%"
                  stopColor="#65FF00"
                  stopOpacity={0.021}
                />
                <stop
                  offset="95%"
                  stopColor="#65FF00"
                  stopOpacity={0.008}
                />
                <stop
                  offset="98%"
                  stopColor="#65FF00"
                  stopOpacity={0.002}
                />
                <stop
                  offset="100%"
                  stopColor="#65FF00"
                  stopOpacity={0}
                />
              </linearGradient>
            </defs>
            <g
              clipPath="url(#victory-clip-2)"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <defs>
                <clipPath
                  id="victory-clip-2"
                >
                  <rect
                    height={300}
                    role="presentation"
                    shapeRendering="auto"
                    vectorEffect="non-scaling-stroke"
                    width={450}
                    x={0}
                    y={0}
                  />
                </clipPath>
              </defs>
              <g>
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150L450,300L425,300C400,300,350,300,300,300C250,300,200,300,150,300C100,300,50,300,25,300L0,300Z"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "url(#-49270000)",
                      "stroke": "none",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "none",
                      "stroke": "#65FF00",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
              </g>
            </g>
          </svg>
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
                "zIndex": 99,
              }
            }
          >
            <svg
              height={300}
              style={
                Object {
                  "height": "100%",
                  "overflow": "visible",
                  "width": "100%",
                }
              }
              viewBox="0 0 450 300"
              width={450}
            />
          </div>
        </div>
      </div>
      <div
        className="large-box__content-delta"
      >
        <div
          className="delta-indicator"
        >
          <span
            className="delta-indicator__delta"
          >
            3pts
          </span>
           
          <span
            className="delta-indicator__arrow delta-indicator__trend-up"
          >
            <svg
              aria-hidden="true"
              color="inherit"
              focusable="false"
              height={5}
              role="presentation"
              viewBox="0 0 10 5"
              width={10}
            >
              <path
                d="M5 0l5 5H0z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <p
    className="large-box__bottom-line"
  />
  <p
    className="large-box__bottom-line"
  >
    Target: 13%
  </p>
</div>
`;

exports[`LargeGraphBox renders PercentageGraphBox with extra content correctly 1`] = `
<div
  className="panel panel--default large-box"
>
  <span
    className="large-box__top-line"
  >
    USV &gt; EXE
  </span>
  <div
    className="large-box__content"
  >
    <span
      className="large-box__content-value"
    >
      10%
    </span>
    <div
      className="large-box__content-extra"
    >
      <div
        className="large-box__content-graph"
      >
        <div
          className="VictoryContainer"
          style={
            Object {
              "height": "100%",
              "pointerEvents": "none",
              "position": "relative",
              "touchAction": "none",
              "width": "100%",
            }
          }
        >
          <svg
            aria-labelledby="victory-container-3-title victory-container-3-desc"
            height={300}
            role="img"
            style={
              Object {
                "height": "100%",
                "pointerEvents": "all",
                "width": "100%",
              }
            }
            viewBox="0 0 450 300"
            width={450}
          >
            <defs>
              <linearGradient
                id="-49270000"
                x1="0%"
                x2="0%"
                y1="0%"
                y2="100%"
              >
                <stop
                  offset="0%"
                  stopColor="#65FF00"
                  stopOpacity={1}
                />
                <stop
                  offset="19%"
                  stopColor="#65FF00"
                  stopOpacity={0.738}
                />
                <stop
                  offset="34%"
                  stopColor="#65FF00"
                  stopOpacity={0.541}
                />
                <stop
                  offset="47%"
                  stopColor="#65FF00"
                  stopOpacity={0.382}
                />
                <stop
                  offset="57%"
                  stopColor="#65FF00"
                  stopOpacity={0.278}
                />
                <stop
                  offset="65%"
                  stopColor="#65FF00"
                  stopOpacity={0.194}
                />
                <stop
                  offset="73%"
                  stopColor="#65FF00"
                  stopOpacity={0.126}
                />
                <stop
                  offset="80%"
                  stopColor="#65FF00"
                  stopOpacity={0.075}
                />
                <stop
                  offset="86%"
                  stopColor="#65FF00"
                  stopOpacity={0.042}
                />
                <stop
                  offset="91%"
                  stopColor="#65FF00"
                  stopOpacity={0.021}
                />
                <stop
                  offset="95%"
                  stopColor="#65FF00"
                  stopOpacity={0.008}
                />
                <stop
                  offset="98%"
                  stopColor="#65FF00"
                  stopOpacity={0.002}
                />
                <stop
                  offset="100%"
                  stopColor="#65FF00"
                  stopOpacity={0}
                />
              </linearGradient>
            </defs>
            <g
              clipPath="url(#victory-clip-4)"
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            >
              <defs>
                <clipPath
                  id="victory-clip-4"
                >
                  <rect
                    height={300}
                    role="presentation"
                    shapeRendering="auto"
                    vectorEffect="non-scaling-stroke"
                    width={450}
                    x={0}
                    y={0}
                  />
                </clipPath>
              </defs>
              <g>
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150L450,300L425,300C400,300,350,300,300,300C250,300,200,300,150,300C100,300,50,300,25,300L0,300Z"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "url(#-49270000)",
                      "stroke": "none",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
                <path
                  d="M0,200L25,183.33333333333334C50,166.66666666666666,100,133.33333333333334,150,100C200,66.66666666666667,250,33.333333333333336,300,41.666666666666664C350,50,400,100,425,125L450,150"
                  role="presentation"
                  shapeRendering="auto"
                  style={
                    Object {
                      "fill": "none",
                      "stroke": "#65FF00",
                      "strokeWidth": "7px",
                      "width": 4,
                    }
                  }
                />
              </g>
            </g>
          </svg>
          <div
            style={
              Object {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
                "zIndex": 99,
              }
            }
          >
            <svg
              height={300}
              style={
                Object {
                  "height": "100%",
                  "overflow": "visible",
                  "width": "100%",
                }
              }
              viewBox="0 0 450 300"
              width={450}
            />
          </div>
        </div>
      </div>
      <div
        className="large-box__content-delta"
      >
        <div
          className="delta-indicator"
        >
          <span
            className="delta-indicator__delta"
          >
            3pts
          </span>
           
          <span
            className="delta-indicator__arrow delta-indicator__trend-up"
          >
            <svg
              aria-hidden="true"
              color="inherit"
              focusable="false"
              height={5}
              role="presentation"
              viewBox="0 0 10 5"
              width={10}
            >
              <path
                d="M5 0l5 5H0z"
                fill="currentColor"
                fillRule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
    </div>
  </div>
  <p
    className="large-box__bottom-line"
  >
    227 Executed Leases (Out of 260)
  </p>
  <p
    className="large-box__bottom-line"
  >
    Target: 13%
  </p>
</div>
`;
